stan/math/prim/mat/err/check_gpu.hpp

```diff
@@ -0,0 +1,272 @@
+        kernel_check_symmetric.setArg(3, buffer_symmetric_flag);
+        kernel_check_symmetric.setArg(4, CONSTRAINT_TOLERANCE);
+
+        cmd_queue.enqueueNDRangeKernel(kernel_check_symmetric,
+         cl::NullRange, cl::NDRange(y.rows(), y.cols()),  cl::NullRange);
+
+        cmd_queue.enqueueReadBuffer(buffer_symmetric_flag, CL_TRUE, 0,
+         sizeof(int), &symmetric_flag);
+        //  if the matrix is not symmetric
+        if (symmetric_flag) {
+          domain_error(function, name, "is not symmetric", "");
+        }
+      } catch (const cl::Error& e) {
+        check_ocl_error("symmetric_check", e);
+      }
+    }
+    /**
+     * Check if the specified matrix on the GPU has zeros on the diagonal
+     *
+     * @param function Function name (for error messages)
+     * @param name Variable name (for error messages)
+     * @param y Matrix to test
+     *
+     * @throw <code>std::domain_error</code> if
+     *    any diagonal element of the matrix is zero.
+     */
+    inline void
+    check_diagonal_zeros(const std::string& function,
+                           const std::string& name,
+                  matrix_gpu& y) {
+      if (y.size() == 0) return;
+
+      cl::Kernel kernel_check_diagonal_zeros =
+       get_kernel("check_diagonal_zeros");
+      cl::CommandQueue cmd_queue = get_queue();
+
+      try {
+        cl::Context ctx = get_context();
+
+        int flag = 0;
+
+        cl::Buffer buffer_flag(ctx, CL_MEM_READ_WRITE,
+         sizeof(int));
+
+        cmd_queue.enqueueWriteBuffer(buffer_flag, CL_TRUE, 0,
+         sizeof(int), &flag);
+
+        kernel_check_diagonal_zeros.setArg(0, y.buffer());
+        kernel_check_diagonal_zeros.setArg(1, y.rows());
+        kernel_check_diagonal_zeros.setArg(2, y.cols());
+        kernel_check_diagonal_zeros.setArg(3, buffer_flag);
+
+        cmd_queue.enqueueNDRangeKernel(kernel_check_diagonal_zeros,
+         cl::NullRange, cl::NDRange(y.rows(), y.cols()),  cl::NullRange);
+
+        cmd_queue.enqueueReadBuffer(buffer_flag, CL_TRUE, 0,
+         sizeof(int), &flag);
+
+        //  if zeros were found on the diagonal
+        if (flag) {
+          domain_error(function, name, "has zeros on the diagonal.", "");
+        }
+      } catch (const cl::Error& e) {
+        check_ocl_error("diag_zeros_check", e);
+      }
+    }
+
+
+
+    /**
+     * Check if the two matrices on the GPU are of the same size.
+     *
+     * This function checks the runtime sizes only.
+     *     
+     *
+     * @param function Function name (for error messages)
+     * @param name1 Variable name for the first matrix (for error messages)
+     * @param y1 First matrix
+     * @param name2 Variable name for the second matrix (for error messages)
+     * @param y2 Second matrix
+     *
+     * @throw <code>std::invalid_argument</code>
+     * if the dimensions of the matrices do not match
+     */
+    inline void check_matching_dims(const std::string& function,
```

Comment:
Using `_gpu` as a suffix should be consistent---either do it everywhere or nowhere.

Reply:
The `_gpu` suffix was removed. The only remaining function with the `_gpu` suffix is the cholesky_decompose_gpu, which can not be overloaded as it expects the same arguments as the existent cholesky_decompose function